lib/sentry-core.module.ts

```diff
@@ -37,7 +38,7 @@ export class SentryCoreModule {
         const provider: Provider = {
             inject: [SENTRY_MODULE_OPTIONS],
             provide: SENTRY_TOKEN,
-            useFactory: (options: SentryModuleOptions) => new SentryService(options),
+            useFactory: (options: SentryModuleOptions) => createSentryService(options),
```

Comment:
I'm a little confused on the injection of the options. It both seems to be injected here via the provider config, as well as in the constructor of the class via the inject-decorator. I would think just the inject-decorator would be infer, and this can be changed to just use `useClass` instead of `useFactory` and remove the `inject` from here.